pkgdb/src/buildenv/assets/mkContainer.nix

```diff
@@ -26,17 +26,22 @@
         (lowPriority containerPkgs.coreutils) # for just the basic utils
       ];
     };
+    # Activate script requires writable /tmp.
+    extraCommands = ''
+      mkdir -m 1777 tmp
+    '';
     config = {
-      # * run -it # [interactive, no args]
-      #   -> runs <Entrypoint> <Cmd>
-      #   -> bash -c -i bash --rcfile <activate>
-      #   (skip activation for the first bash and runs default rcfiles)
-      #
-      # * run cmd... # [non-interactive, with arguments]
-      #   -> BASH_ENV=<activate> bash -c cmd
-
-      # * follow convention of sh -c being container entrypoint
-      Entrypoint = ["${containerPkgs.bashInteractive}/bin/bash" "-c"];
+      # Use activate script as the [one] entrypoint capable of
+      # detecting interactive vs. command activation modes.
+      # Usage:
+      #   podman run -it
+      #     -> launches interactive shell with controlling terminal
+      #   podman run -i <cmd>
+      #     -> invokes interactive command
+      #   podman run -i [SIC]
+      #     -> launches crippled interactive shell with no controlling
+      #        terminal .. kinda useless
+      Entrypoint = ["${environment}/activate"];
```

Comment:
From a brief functional review this was doing what I expect. It doesn't follow the "convention of sh -c being container entrypoint" but I just looked at debian and it doesn't have that as entrypoint, and afaict the docker CLI doesn't default to that either